### PR TITLE
Update Configuration Reference Documentation to use @ConfigMapping

### PR DESCRIPTION
**Pull request by radcortez**

### CODE REVIEW

File: docs/src/main/asciidoc/writing-extensions.adoc [hunk 1]
```diff
@@ -1201,14 +1201,200 @@ quarkus.log.file.level=DEBUG
...
+
+While standard Javadoc comments are perfectly fine for simple documentation (recommended even), Asciidoc is more
+suitable for tips, source code extracts, lists and more:
+
+[source,java]
+----
+/**
+ * Class name of the Hibernate ORM dialect. The complete list of bundled dialects is available in the
+ * https://docs.jboss.org/hibernate/stable/orm/javadocs/org/hibernate/dialect/package-summary.html[Hibernate ORM JavaDoc].
+ * <br>
```
Comment (gsmet) [inline docs/src/main/asciidoc/writing-extensions.adoc:1238]: You shouldn't have a `<br>` here as it's AsciiDoc.

File: docs/src/main/asciidoc/writing-extensions.adoc [hunk 2]
```diff
@@ -1201,14 +1201,200 @@ quarkus.log.file.level=DEBUG
...
+===== Writing the documentation
+
+Each configuration property, requires a Javadoc explaining its purpose.
+
+[TIP]
+====
+The first sentence should be meaningful and self-contained as it is included in the summary table.
+====
+
+While standard Javadoc comments are perfectly fine for simple documentation (recommended even), Asciidoc is more
```
Comment (gsmet) [inline docs/src/main/asciidoc/writing-extensions.adoc:1230]:
```suggestion
While standard Javadoc comments are perfectly fine for simple documentation (recommended even), AsciiDoc is more
```

File: docs/src/main/asciidoc/writing-extensions.adoc [hunk 3]
```diff
@@ -1201,14 +1201,200 @@ quarkus.log.file.level=DEBUG
...
+ * Not all the dialects are supported in GraalVM native executables: we currently provide driver extensions for
+ * PostgreSQL, MariaDB, Microsoft SQL Server and H2.
+ * ====
+ *
+ * @asciidoclet
+ */
+Optional<String> dialect();
+----
+
+To use Asciidoc, the Javadoc comment must be annotated with `@asciidoclet` tag. This tag serves two purposes: it is
```
Comment (gsmet) [inline docs/src/main/asciidoc/writing-extensions.adoc:1250]:
```suggestion
To use AsciiDoc, the Javadoc comment must be annotated with `@asciidoclet` tag. This tag serves two purposes: it is
```

File: docs/src/main/asciidoc/writing-extensions.adoc [hunk 4]
```diff
@@ -1201,14 +1201,200 @@ quarkus.log.file.level=DEBUG
...
+----
+
+For indentation to be respected in the Javadoc comment (list items spread on multiple lines or indented
+source code), the automatic Eclipse formatter must be disabled (the formatter is automatically included in the build),
+with the markers `// @formatter:off`/`// @formatter:on`. These require separate comments and a mandatory space after the
+`//` marker.
+
+[WARNING]
+====
+Open blocks (`--`) are not supported in the Asciidoctor documentation. All the other types of blocks
```
Comment (gsmet) [inline docs/src/main/asciidoc/writing-extensions.adoc:1300]:
```suggestion
Open blocks (`--`) are not supported in the AsciiDoc documentation. All the other types of blocks
```

File: docs/src/main/asciidoc/writing-extensions.adoc [hunk 5]
```diff
@@ -1201,14 +1201,200 @@ quarkus.log.file.level=DEBUG
...
+----
+<1> This will add a section documentation for the `configGroup` config item in the generated documentation. The section
+title and introduction will be derived from the javadoc of the configuration item. The first sentence from the javadoc
+is considered as the section title and the remaining sentences used as section introduction.
+
+===== Generating the documentation
+
+To generate the documentation:
+
+* Execute `./mvnw clean install -DskipTests -DskipITs`
```
Comment (gsmet) [inline docs/src/main/asciidoc/writing-extensions.adoc:1346]:
```suggestion
* Execute `./mvnw -DquicklyDocs`
```

**Review by gsmet — CHANGES_REQUESTED** [review]
I think you probably just moved a paragraph but I noticed that too late :)... So here you are with a couple of suggestions.

**Review by gsmet — APPROVED** [review]
I will merge this and apply the changes myself as it's not fair to you and you probably have a lot on your plate this week!

**Comment by radcortez** [comment]
No worries, I could have done that.

**Pull request merged**